MIGRATION.md

```diff
@@ -0,0 +1,193 @@
+
+##### `ripple-binary-codec`
+
+- `SerializedType` constructor and `toBytes` .  Its sub-classes:
+  - `AccountID`
+  - `Amount`
+  - `Blob`
+  - `Currency`
+  - `Hash`
+  - `Hash128`
+  - `Hash160`
+  - `Hash256`
+  - `Issue`
+  - `PathSet`
+  - `STArray`
+  - `STObject`
+  - `UInt`
+  - `UInt8`
+  - `UInt16`
+  - `UInt32`
+  - `UInt64`
+  - `Vector256`
+  - `XChainBridge`
+- `ShaMapNode.hashPrefix`
+- `BinarySerializer.put`
+- `BytesList.put` and `BytesList.toBytes`
+- `BinaryParser.read`
+- `BinaryParser.readVariableLength`
+- `Quality.encode` and `Quality.decode`
+- `Sha512Half.put` and `Sha512Half.finish256`
+- `transactionID`
+- `sha512Half`
+- Entries of `HashPrefix`
+- Type `FieldInstance.header`
+- `Bytes.bytes`
+- `signingClaimData`
+- `serializeObject`
+- `makeParser`
+
+##### `secret-numbers`
+
+- `entropyToSecret`
+- `randomEntropy`
+- `Account` constructor
+
+##### `xrpl`
+
+- `rfc1751MnemonicToKey`
+
+### Large Number Handling
+
+`bn.js`, `decimal.js` and `big-integer` were removed as dependencies.  They usages were replaced with `BigNumber` from `big-number.js` (was already a dependency) and the native javascript object `BigInt`.
+
+- `UInt64.valueOf` returns `bigint` instead of  `BigInteger`
+- `SerializeType.from` can take a  `bigint` instead `BigInteger`
+- `ledgerHash` had its param object change so that `total_coins` in a `bigint` instead `BigInteger`
+- `Quality.decode` returns a `BigNumber` instead of a `Decimal`
+- `Amount.assertIouIsValid` take a `BigNumber` instead `Decimal`
+- `Amount.verifyNoDecimal` takes a `BigNumber` instead `Decimal`
+
+### Error Handling
+
+In order to remove `assert` as a dependency regular `Error` objects are thrown instead of  `AssertionError`.
+
+Not a breaking change but there is now a much more descriptive error message when keys do not match for a given algorithm.  This impacts most of `ripple-keypairs` functions but only if you already had issues with incompatible values.
+
+### Bundlers
+
+For `vite` and `create-react-app` you can remove all xrpl.js polyfills/configurations this also includes the custom mappings for `ws` to `WsWrapper` and the exclusion of `https-proxy-agent`. These improves should extend to using the libraries with other bundlers or frameworks but has only been tested in those noted above.
+
+#### React Native
+
+Neither `react-native-crypto` nor `rn-nodeify` are required.  Instead `fast-text-encoding` and `react-native-get-random-values` are as well as a small polyfills file to wire them up. You can find the new guide at [UNIQUE_SETUPS.md](./UNIQUE_SETUPS.md).
+
+### `xrpl-secret-numbers`
+
+This library was absorbed into the monorepo as it was causing dependency mismatches and was going to require an update anyway.  It is now available as `@xrplf/secret-numbers`.  Besides making changes to use `UInt8Array` you will need to update all places where you use functions on the `Util` object.  These methods are now at the root of the library. These methods include:
```

Comment:
Is the `Util` object specific to `xrpl-secret-numbers` or should this be it's own section?